libs/features/advert-form/src/lib/features-advert-form.tsx

```diff
@@ -55,45 +56,50 @@ export const FeaturesAdvertForm = (props: RouteComponentProps<RouteInfo>) => {
     onSubmit: create,
     onCancel: redirect,
     data: {
-      title: 'New advert',
-      submitButtonText: 'Create',
-      descriptionInputLabel: 'Description',
-      titleInputLabel: 'Title',
-      categoryInputLabel: 'Category',
-      cancelButtonText: 'Cancel',
+      id: advert.id,
+      title: advert.string,
+      description: advert.description,
+      advertImage: advert.FileList,
+      category: advert.category,
+      prerequisites: advert.prerequisites,
+      learning: advert.learning,
+      lessons: advert.lessons,
     },
-    categories,
+    categories: categories,
+    advert,
   };
 
   const editContent = {
     onSubmit: update,
     onCancel: redirect,
     data: {
-      title: 'Edit advert',
-      submitButtonText: 'Update',
-      descriptionInputLabel: 'Description',
-      titleInputLabel: 'Title',
-      categoryInputLabel: 'Category',
-      cancelButtonText: 'Cancel',
+      id: advert.id,
+      title: advert.string,
+      description: advert.description,
+      advertImage: advert.FileList,
+      category: advert.category,
+      prerequisites: advert.prerequisites,
+      learning: advert.learning,
+      lessons: advert.lessons,
     },
     categories: categories,
     advert,
   };
 
-  const [content, setContent] = useState(createContent);
+  const [content, setContent] = useState<SharedCreateAdvertFormProps>(createContent);
 
   useEffect(() => {
     props.match.path ===
       `/${PrivateRoutesPath.ADVERTS}${PrivateRoutesPath.GET_BY_ID}/edit` &&
       dispatch(advertsActions.getById(props.match.params.id));
     dispatch(categoriesActions.getAll());
-  }, []);
+  }, [dispatch, props.match.params.id, props.match.path]);
```

Comment:
revert, this triggers [dispatch, props.match.params.id, props.match.path] is useless, because it trigger only on init, is same as empty array;